core/frontend/src/render/webgl/AtmosphericScatteringUniforms.ts

```diff
@@ -0,0 +1,314 @@
+  private _isCameraEnabled = true;
+  private _minDensityScaleMatrix = new Matrix3d(new Float64Array([1,0,0,0,1,0,0,0,1]));
+  private _minDensityToAtmosphereScaleFactor: any;
+  private _numInScatteringPoints = 0.0;
+  private _numOpticalDepthPoints = 0.0;
+  private _outScatteringIntensity = 1.0;
+  private _scatteringCoefficients = new Float32Array(3);
+  private readonly _ellipsoidRotationMatrix = new Matrix3d();
+  private _isEnabled = false;
+
+  // utility
+  public syncKey = 0;
+  private _scratchVector3d = new Vector3d();
+  private _scratchMatrix3d = new Matrix3d();
+
+  public get atmosphericScattering(): AtmosphericScattering.Settings | undefined {
+    return this._atmosphericScattering;
+  }
+
+  public update(target: Target): void {
+    desync(this);
+
+    const plan = target.plan;
+    if (!(this.atmosphericScattering && plan.atmosphericScattering && this.atmosphericScattering.equals(plan.atmosphericScattering))) {
+      this._atmosphericScattering = plan.atmosphericScattering;
+    }
+
+    this._updateIsEnabled(target.wantAtmosphericScattering);
+
+    if (!this.atmosphericScattering) {
+      return;
+    }
+
+    this._updateAtmosphereHeightAboveEarth(this.atmosphericScattering.atmosphereHeightAboveEarth);
+    this._updateAtmosphereScaleMatrix(this.atmosphericScattering.atmosphereHeightAboveEarth);
+    this._updateBrightnessAdaptationStrength(this.atmosphericScattering.brightnessAdaptationStrength);
+    this._updateDensityFalloff(this.atmosphericScattering.densityFalloff);
+    this._updateEarthCenter(plan.ellipsoidCenter!, target.uniforms.frustum.viewMatrix);
+    this._updateEarthScaleMatrix(plan.ellipsoidRadii!);
+    this._updateEllipsoidRotationMatrix(plan.ellipsoidRotation!);
+    this._updateEllipsoidToEye();
+    this._updateInScatteringIntensity(this.atmosphericScattering.inScatteringIntensity);
+    this._updateInverseEllipsoidRotationMatrix(plan.ellipsoidRotation!, target.uniforms.frustum.viewMatrix.matrix);
+    this._updateIsCameraEnabled(target.uniforms.frustum.type);
+    this._updateMinDensityScaleMatrix(this.atmosphericScattering.minDensityHeightBelowEarth);
+    this._updateMinDensityToAtmosphereScaleFactor(this.atmosphericScattering.atmosphereHeightAboveEarth, this.atmosphericScattering.minDensityHeightBelowEarth);
+    this._updateNumInScatteringPoints(this.atmosphericScattering.numInScatteringPoints);
+    this._updateNumOpticalDepthPoints(this.atmosphericScattering.numOpticalDepthPoints);
+    this._updateOutScatteringIntensity(this.atmosphericScattering.outScatteringIntensity);
+    this._updateScatteringCoefficients(this.atmosphericScattering.scatteringStrength, this.atmosphericScattering.wavelengths);
+  }
+
+  private _updateIsEnabled(wantAtmosphericScattering: boolean) {
+    this._isEnabled = wantAtmosphericScattering;
+  }
+
+  private _updateIsCameraEnabled(frustumType: FrustumUniformType) {
+    this._isCameraEnabled = frustumType === FrustumUniformType.Perspective;
```

Comment:
We already have a uniform for this that is possibly already being used by these shaders - see `FrustumUniforms`.